### DISCUSSION TITLE
FATAL ERROR: while building the Dashboard

### DISCUSSION THREAD

**Discussion by onelsonic** [saleor/saleor#5964] [Q&A] [answered]
Hi there,
I keep getting the following error on a machine : when trying to build the Dashboard :

`FATAL ERROR: Ineffective mark-compacts near heap limit Allocation failed - JavaScript heap out of memory`

this seems to be a nodejs error saying there is not enough memory? Anyone encounter this issue?
The machine has 2GB of memory, is there any memory restrictions in order to build the Dashboard? (if yes it should me mentioned somewhere...)
Any chance to bypass this error?

Otherwise can I build the Dashboard files locally, to then deploy them online?
Thanks

**Comment by nalimhahs** [comment]
Not sure about the error but you can build it locally and then deploy them online.

**Comment by neerajgupta2407** [comment, marked as answer]
@onelsonic While building dashboard build, memory spikes to approx. 3 gb. 
So make sure to have 4gb available memory while building dashboard.

or as you have mentioned, you can build locally with appropriate environment variables and deployed to your servers.

**Comment by onelsonic** [comment]
Hi there,
thank you for the reply
I built the dashboard locally then deployed it, it works. But unable to built it with only 2GB of ram.

**Reply by mitraxis** [reply]
Did you manage to solve this issue?

**Reply by onelsonic** [reply]
yes and you will need more than 2GB of ram in order to build it.
